src/box/tuple.c

```diff
@@ -796,6 +800,16 @@ box_tuple_validate(box_tuple_t *tuple, box_tuple_format_t *format)
 
 /* }}} box_tuple_* */
 
+box_tuple_t *
+box_tuple_new_with_validation(box_tuple_format_t *format, const char *data,
+			      const char *end, bool validate)
+{
+	struct tuple *ret = tuple_new(format, data, end, validate);
+	if (ret == NULL)
+		return NULL;
```

Comment:
```
tarantool> box.tuple.new({1, 2, 3}, {format = {{type = 'integer', name = 'id'}, {type = 'integer', name = 'id2', foreign_key = {city = {space = 'city', field = 'id'}}}}})
---
- error: "builtin/box/load_cfg.lua:985: Please call box.cfg{} first\nstack traceback:\n\tbuiltin/box/load_cfg.lua:985:
    in function '__index'\n\tbuiltin/box/schema.lua:622: in function 'normalize_foreign_key_one'\n\tbuiltin/box/schema.lua:691:
    in function 'normalize_foreign_key'\n\tbuiltin/box/schema.lua:729: in function
    'normalize_format'\n\tbuiltin/box/tuple.lua:405: in function 'new'\n\tbuiltin/box/tuple.lua:89:
    in function <builtin/box/tuple.lua:70>\n\t[C]: in function 'pcall'\n\tbuiltin/box/console.lua:427:
    in function 'eval'\n\tbuiltin/box/console.lua:819: in function 'repl'\n\tbuiltin/box/console.lua:870:
    in function <builtin/box/console.lua:858>"
...
```

I even don't know that should we do in such case. But my suggestion is to ignore anything except name, type, is_nullable in format

Reply:
I am reusing the format normalization function we use for `space:format`, hence the error. Actually the format API states we only support 'name' and 'type' fields, so passing anything else is kind of UB. 

I would prefer not dealing with this (i.e., not writing a new format normalization function for `box.tuple.format`) — but it other reviewers insist, I will.

Reply:
See [comment](https://github.com/tarantool/tarantool/pull/8549#discussion_r1173998353).

Reply:
Also, considering this particular issue, we discussed it with @locker and @alyapunov and came to a conclusion that it's okay we throw this error — if one wants to use collations, constraints or foreign keys in tuple format, one is aware it requires box to be configured.